builtin/providers/azurerm/resource_arm_availability_set.go

```diff
@@ -0,0 +1,140 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAvailabilitySet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAvailabilitySetCreate,
+		Read:   resourceArmAvailabilitySetRead,
+		Update: resourceArmAvailabilitySetCreate,
+		Delete: resourceArmAvailabilitySetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"platform_update_domain_count": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  5,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(int)
+					if value > 20 {
+						errors = append(errors, fmt.Errorf(
+							"%q maximum of 20 update domains can be used", k))
```

Comment:
This will probably read somewhat strangely? `"platform_update_domain_count" maximum of 20 update domains can be used` - we might want to consider rewording it to something like `Maximum value for "platform_update_domain_count" is 20`?